src/io/msh_basereader.cc

```diff
@@ -0,0 +1,137 @@
+/*!
+ *
+﻿ * Copyright (C) 2015 Technical University of Liberec.  All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 3 as published by the
+ * Free Software Foundation. (http://www.gnu.org/licenses/gpl-3.0.en.html)
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+ *
+ *
+ * @file    msh_basereader.cc
+ * @brief
+ * @author  dalibor
+ */
+
+
+#include "io/msh_basereader.hh"
+#include "io/msh_gmshreader.h"
+#include "io/msh_vtkreader.hh"
+#include "system/sys_profiler.hh"
+
+
+BaseMeshReader::BaseMeshReader(const Input::Record &mesh_rec)
+: tok_(mesh_rec.val<FilePath>("mesh_file")) {
+	input_mesh_rec_ = mesh_rec;
+}
+
+BaseMeshReader::BaseMeshReader(std::istream &in)
```

Comment:
- have just single constructor which takes stream or rather FilePath as parameter
- static method `reader factory` can be used to construct reader depending on the extension
- static method `mesh_factory` can be used to construct reader and call read_mesh, the mesh input record is passed to it and to `read_mesh` as well